subworkflows/local/hic_mapping.nf

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env nextflow
+
+// This subworkflow takes an input fasta sequence and csv style list of hic cram file to return
+// alignment files including .mcool, pretext and .hic.
+// Input - Assembled genomic fasta file, cram file directory
+// Output - .mcool, .pretext, .hic
+
+//
+// MODULE IMPORT BLOCK
+//
+include { BWAMEM2_INDEX                             } from '../../modules/nf-core/bwamem2/index/main'
+include { COOLER_CLOAD                              } from '../../modules/nf-core/cooler/cload/main'
+include { COOLER_ZOOMIFY                            } from '../../modules/nf-core/cooler/zoomify/main'
+include { PRETEXTMAP as PRETEXTMAP_LOWRES           } from '../../modules/nf-core/pretextmap/main'
+include { PRETEXTMAP as PRETEXTMAP_HIGHRES          } from '../../modules/nf-core/pretextmap/main'
+include { SAMTOOLS_MARKDUP                          } from '../../modules/nf-core/samtools/markdup/main'
+include { SAMTOOLS_MERGE                            } from '../../modules/nf-core/samtools/merge/main'
+include { BAMTOBED_SORT                             } from '../../modules/local/bamtobed_sort.nf'
+include { GENERATE_CRAM_CSV                         } from '../../modules/local/generate_cram_csv'
+include { CRAM_FILTER_ALIGN_BWAMEM2_FIXMATE_SORT    } from '../../modules/local/cram_filter_align_bwamem2_fixmate_sort'
+include { JUICER_TOOLS_PRE                          } from '../../modules/local/juicer_tools_pre'
+include { GET_PAIRED_CONTACT_BED                    } from '../../modules/local/get_paired_contact_bed'
+
+
+workflow HIC_MAPPING {
+    take:
+    reference_tuple     // Channel [ val(meta), path(file) ]
+    reference_index     // Channel [ val(meta), path(file) ]
+    dot_genome          // Channel: [val(meta), [ datafile ]]
+    hic_reads_path      // Channel [ val(meta), path(directory) ]
+
+    main:
+    ch_versions         = Channel.empty()
+
+    ch_cool_bin         = Channel.of(1)
```

Comment:
The 1 tells cooler to use the values derived from the .genome file rather than rely on given intervals of bp. This could be injected via map if you think that's more suitable. Either way I'll add a comment.

Reply:
https://cooler.readthedocs.io/en/latest/cli.html#cooler-cload

Reply:
All good. Thanks for explaining.